sycl/doc/design/OptionalDeviceFeatures.md

```diff
@@ -423,6 +423,20 @@ allow metadata to be attached directly to types.  This representation works
 around that limitation by creating global named metadata that references the
 type's name.
 
+To synchronize the integral values of given aspects between the SYCL headers and
+the compiler, the `!intel_sycl_aspects` metadata is added to the module, based
+on the values defined in the enum. Inside this metadata node, each value of the
+aspect enum is represented by another metadata node with two operands; the name
+of the value and the corresponding integral value. An example of this is:
+
+```
+!intel_sycl_aspects = !{!0, !1, !2, ...}
+!0 = !{!"host", i32 0}
+!1 = !{!"cpu", i32 1}
+!2 = !{!"gpu", i32 2}
+...
+```
```

Comment:
It's not very clear from this design document how the `intel_sycl_aspects` metadata is used.  I think the reason is to provide the enum value for `aspect::fp64` to the new LLVM IR pass, correct?  If that is the case, can you add a note to the description of that pass saying that it uses this metadata for that purpose?

Did you consider as an alternate solution to have the CFE include one more entry in `intel_types_that_use_aspects`?  For example:

```
!intel_types_that_use_aspects = !{!0, ...}
!0 = !{!"double", i32 8}  # The value 8 is "aspect::fp64"
```